test.py

```diff
@@ -478,12 +479,55 @@ def build_test_list(self) -> List[str]:
         tests = [os.path.splitext(t.relative_to(self.suite_path))[0] for t in pytests]
         return self._filter_test_list(tests)
 
+    @property
+    def _extra_pytest_dummy_args(self) -> List[str]:
+        # Needed args for listing test cases
+        return []
+
+    async def _pytest_list_cases(self, test_file: str) -> List[Optional[str]]:
```

Comment:
Why do you need to list the cases, if anyway the UI is to run one specific case and not some glob like "all tests beginning with the letter x"?

Reply:
The current semantics is to actually run all matches, so that you can type ./test.py raft and that would run *everything* in *all suites* that has name "raft" in it.

Reply:
What would you think of `./test.py raft` for matching any and `./test.py raft::` for matching the suite raft? We need to make it unambiguous.

Reply:
Exactly. It is working like this today. The match is made against a fully qualified test name, i.e. suite_name.test_file_name. After your change, it will be against a fully qualified test name as well, but the syntax will be suite_name::test_file_name::test_case_name. But it's still going to be pattern match, not exact match, so it'll be fine.